A path named twice in versionedFiles makes createCommitOnBranch refuse the release commit

Since v7.9.0, a release that bumps a package whose `versionedFiles` name the same path more than once fails at the commit:

```
error[E3014]: createCommitOnBranch failed: Variable $input of type CreateCommitOnBranchInput! was provided invalid value for fileChanges (Paths must be unique across additions and deletions. At least one path was included more than once.)
```

Naming the same path more than once is a supported and documented shape: one entry per selector, which is how a Helm chart bumps `version`, `appVersion` and an image line in one `Chart.yaml`. LFSX does exactly that, three entries, and its server release has shipped this way since 2026-08-18, most recently v1.5.0 on 08-25 under v7.7.1. Today's release under v7.9.0 failed with the error above (LFSX run 33148431181).

## What changed

#941 made the release commit go through `createCommitOnBranch` so it is authored by ferrflow[bot] and verified. Before that, the bump paths went to `git add`, where a duplicate is harmless, which is why the duplicate never surfaced.

The duplicate itself comes from the bump loop, `src/monorepo/run/mod.rs:550`: `files_to_commit.push(vf.path.clone())` runs once per `versionedFiles` entry, not once per file. `refresh_lockfiles`, a few lines above, already guards the same push with a `handled` set, so the care exists in the file, just not here.

## Where the fix belongs

`file_changes` in `src/monorepo/run/authored_commit.rs`, because it is the one place that feeds an API which requires uniqueness, and it protects every caller including any future one. The contents are read from disk per path, so duplicates are byte-identical and keeping the first is correct. Deduplicating at the bump-loop push would fix this instance and leave the API boundary trusting its input.

Both `additions` and `deletions` need it: a path listed twice that no longer exists on disk becomes two deletions.

--- a/src/monorepo/run/authored_commit.rs
+++ b/src/monorepo/run/authored_commit.rs
@@ -11,8 +11,18 @@ pub(super) fn file_changes(
 ) -> Result<(Vec<FileAddition>, Vec<String>)> {
     let mut additions = Vec::new();
     let mut deletions = Vec::new();
+    let mut seen = std::collections::HashSet::new();
 
+    // The bump loop pushes a path once per versionedFiles entry, and a file
+    // carrying several selectors is a supported shape, so the same path
+    // arrives here more than once. `git add` never minded; the API refuses
+    // the whole commit over it. The contents are read from disk per path, so
+    // the copies are identical and the first one is the file.
     for file in files {
+        if !seen.insert(*file) {
+            continue;
+        }
+
         let path = root.join(file);
         if path.exists() {
             let bytes = std::fs::read(&path).with_context(|| {
@@ -101,6 +111,32 @@ mod tests {
             .unwrap();
 
         assert_eq!(decoded, bytes);
+    }
+
+    // Three selectors on one Chart.yaml is how a Helm chart bumps its
+    // version, appVersion and image line, and the API refuses a commit whose
+    // fileChanges name a path twice. This is the shape that failed LFSX's
+    // v1.5.1 release under v7.9.0.
+    #[test]
+    fn a_path_named_by_several_entries_is_sent_once() {
+        let dir = tempfile::tempdir().unwrap();
+        std::fs::write(dir.path().join("Chart.yaml"), "version: 1.2.3").unwrap();
+
+        let (additions, deletions) = file_changes(
+            dir.path(),
+            &[
+                "Chart.yaml",
+                "Chart.yaml",
+                "Chart.yaml",
+                "gone.txt",
+                "gone.txt",
+            ],
+        )
+        .unwrap();
+
+        assert_eq!(additions.len(), 1);
+        assert_eq!(additions[0].path, "Chart.yaml");
+        assert_eq!(deletions, vec!["gone.txt".to_string()]);
     }
 
     #[test]

